parachain/pallets/verifier/src/lib.rs

```diff
@@ -93,12 +156,71 @@ impl<T: Trait> Module<T> {
 	}
 
 	// Verify that the message sender matches the relayer account
-	fn verify_sender(sender: T::AccountId) -> DispatchResult {
-		if sender != RelayKey::<T>::get() {
+	fn verify_sender(sender: &T::AccountId) -> DispatchResult {
+		if *sender != RelayKey::<T>::get() {
 			return Err(Error::<T>::Invalid.into())
 		}
 		Ok(())
 	}
+
+	// Validate an Ethereum header for import
+	fn validate_header_to_import(sender: &T::AccountId, header: &EthereumHeader) -> Result<(), &'static str> {
```

Comment:
The plan is to return `ValidTransaction` struct here in future. So instead of calling this in `import_header`, have the transaction pool call this and use the `ValidTransaction` to order / drop headers. Then I can remove the parent-before-child check below